src/server/mgflask/mgflask/news.py

```diff
@@ -1,19 +1,60 @@
 from flask import (
     Blueprint,
     request,
-    jsonify
+    jsonify,
+    request
 )
-from newsapi import NewsApiClient
+from newsapi import NewsApiClient 
 
 bp = Blueprint('news', __name__, url_prefix='/news')
 
+api_keys = ['983d4d9ce3dc4f3badda1a1171eb548d', 'b5ad966ba07741858c365a83ed18a0bb']
+target_sources = ["bbc-news", "fox-news", "the-wall-street-journal", "national-review", "the-huffington-post", "the-hill", "cnn"]
+accepted_params= [ 'sources','qintitle','q','domains','excludeDomains', 'from', 'to', 'language', 'sortBy', 'page', 'pageSize']
 
-@bp.route('/asdf', methods=['GET'])
-def testNewsAPI():
-    newsapi = NewsApiClient(api_key='983d4d9ce3dc4f3badda1a1171eb548d')
-    top_headlines = newsapi.get_everything(q='bitcoin',
-                                           sources='bbc-news,the-verge',
-                                           domains='bbc.co.uk,techcrunch.com',
-                                           language='en',
-                                           sort_by='relevancy')
-    return jsonify(top_headlines)
+
+#get headlines from all target sources
+@bp.route('/headlines', methods=['GET'])
+def get_headlines_from_all():
+    topheadlines = {}
+    newsapi = NewsApiClient(api_key=api_keys[0])
+    sources = ",".join(target_sources)
+    print(str(request_params))
+    try:
+      topheadlines = newsapi.get_top_headlines(sources=sources)
+    except ValueError as e:
+      topheadlines['status'] = 'error'
+      topheadlines['message'] = str(e)
+
+    return jsonify(topheadlines)
+
+#get headlines from a designated target source
+@bp.route('/headlines/<news_source>', methods=['GET'])
+def get_headlines_from(news_source):
+    newsapi = NewsApiClient(api_key=api_keys[0])
+    try:
+      topheadlines = newsapi.get_top_headlines(sources=news_source)
+    except ValueError as e:
+      topheadlines['status'] = 'error'
+      topheadlines['message'] = str(e)
+
+    return jsonify(topheadlines)
+
+#search among all articles
+@bp.route('/everything', methods=['GET'])
```

Comment:
consider changing '/everything' to just '/' since we'd prob want to grab all the articles when the user first opens it

Reply:
sounds good